src/rcc.rs

```diff
@@ -478,28 +470,272 @@ impl CFGR {
+                    }
+                } else {
+                    n -= 1;
+                    continue;
+                }
+            }
+
+            if let Some((q_freq_min, q_freq_max)) = freq_req.q {
+                let mut div = None;
+                for div_q in 2..=15 {
+                    let f_usb_clock_output = f_vco_clock / div_q;
+                    if f_usb_clock_output >= q_freq_min && f_usb_clock_output <= q_freq_max {
+                        div = Some(div_q)
+                    }
+                }
+                if div.is_some() {
+                    q = div;
+                    break;
+                } else {
+                    n -= 1;
+                    continue;
+                }
+            }
+        }
+
+        Some((m, n, p, q))
+    }
+
+    fn pll_configure(mut self) -> Self {
+        let base_clk = match self.hse.as_ref() {
+            Some(hse) => hse.freq,
+            None => HSI,
+        };
+
+        let sysclk = if let Some(clk) = self.sysclk {
+            clk
+        } else {
+            base_clk
+        };
+
+        let p = if base_clk == sysclk {
+            None
+        } else {
+            Some((sysclk - 1, sysclk + 1))
+        };
+
+        let q = if self.use_pll48clk {
+            Some((48_000_000 - 120_000, 48_000_000 + 120_000))
+        } else {
+            None
+        };
+
+        if let Some((m, n, p, q)) = CFGR::calculate_mnpq(base_clk, FreqRequest { p, q }) {
+            self.pllm = m as u8;
+            self.plln = n as u16;
+            if let Some(p) = p {
+                self.use_pll();
+                self.pllp = match p {
+                    2 => PLLP::Div2,
+                    4 => PLLP::Div4,
+                    6 => PLLP::Div6,
+                    8 => PLLP::Div8,
+                    _ => unreachable!(),
+                };
+            }
+            if let Some(q) = q {
+                self.pllq = q as u8;
+            }
+        } else {
+            panic!("couldn't calculate {} from {}", sysclk, base_clk);
+        }
+
+        self
+    }
+
+    /// Configure the default clock settings.
+    /// Set sysclk as 216 Mhz and setup USB clock if defined.
+    pub fn set_defaults(self) -> Self {
+        self.sysclk(Hertz(216_000_000));
+
+        self
+    }
+
+    /// Configure the "mandatory" clocks (`sysclk`, `hclk`, `pclk1` and `pclk2')
+    /// and return them via the `Clocks` struct.
+    ///
+    /// The user shouldn't call freeze more than once as the clocks parameters
+    /// cannot be changed after the clocks have started.
+    ///
+    /// The implementation makes the following choice: HSI is always chosen over
+    /// HSE except when HSE is provided. When HSE is provided, HSE is used
+    /// wherever it is possible.
+    pub fn freeze(self) -> Clocks {
+        let flash = unsafe { &(*FLASH::ptr()) };
+        let rcc = unsafe { &(*RCC::ptr()) };
+        let pwr = unsafe { &(*PWR::ptr()) };
+
+        self.pll_configure();
```

Comment:
`pll_configure` is Self->Self, so this operation is no-op.

Reply:
Fixed